proxy-readme.md

```diff
@@ -0,0 +1,148 @@
+# Application Builder Proxy
+
+The Application Builder Proxy is a thin server side component that serves two important purposes.
+
+1. The proxy is an abstraction between Application Builder widgets and web services that are external to Engine.  Calling the proxy instead of directly calling an external web service from widgets promotes better maintainability.  For example, when an external endpoint changes only the proxy must be updated rather than the URL references in individual widgets.
+
+2. The proxy enables client-side interaction with external services via Ajax calls in the browser. Modern browser security prevents cross site Ajax requests due to the same-origin policy.
+
+[Watson Explorer Application Builder 11 and later includes a feature called Endpoints](http://www-01.ibm.com/support/knowledgecenter/SS8NLW_11.0.0/com.ibm.swg.im.infosphere.dataexpl.appbuilder.doc/c_de-ab-devapp-endpoints.html) which serves this purpose.  A proxy application like this one would not be needed if you are using version 11 of the software or later.
+
+
+# Installing the Sample Proxy
+
+$AB_HOME refers to the home folder for the specific Application Builder installation.  For example, on a default Windows installation, `$AB_HOME` might be C:\Program Files\IBM\WEX\AppBuilder
+
+Create a new folder called 'proxy' in `$AB_HOME/wlp/usr/servers/AppBuilder/apps`
+
+Unzip the proxy WAR into the newly created proxy folder.  You should now have `$AB_HOME/wlp/usr/servers/AppBuilder/apps/proxy/META-INF` and `$AB_HOME/wlp/usr/servers/AppBuilder/apps/proxy/WEB-INF`.
+
+Open server.xml, located at $AB_HOME/wlp/usr/servers/AppBuilder/ for editing.  Add the following block to the `server` node.
+
+
+```xml
+   <application type="war" id="proxy" name="proxy" location="${server.config.dir}/apps/proxy">
+   </application>
+```
+
+
+Before using the Sample Proxy it is necessary to update the proxy configuration to do something interesting.  For example, you might add functionality that points to custom [Watson Developer Cloud](https://console.ng.bluemix.net/) applications. 
+
+Once you've modified the proxy to your liking, restart Application Builder's WebSphere according to the App Builder documentation - by restarting the Application Builder service using the normal commands for your installation.
+
+
+# Usage
+
+The sample proxy includes a single endpoint out of the box for `ping`.
+
+```
+curl http://localhost:9080/proxy/ping
+```
+
+You can also point your browser at [http://localhost:9080/proxy/ping](http://localhost:9080/proxy/ping). Change the port or hostname as appropriate specific to your sever.
```

Comment:
Note that this port number must change if the administrator has chosen a different appbuilder port.